### DISCUSSION TITLE
MNE compatible leadfield with SimNIBS

### DISCUSSION THREAD

**Discussion by anujanegi** [simnibs/simnibs#180] [7.Creating head models]
Hi,
My main objective is to create a leadfield using SimNIBS and then downsample it to an atlas space.

**Data:**
T1 and T2 scans.

**Steps I think I can do to achieve this:**
1. Create head model using charm.
2. Run TDCS leadfield calculation.
3. Convert this to MNE python format.
4. Convert leadfield from free to fixed orientation.
5. Simple downsampling by adding voxels that belong to the same regions in the atlas.

**Current issue:**
I'm at step 3, and trying to follow instructions from [this](https://doi.org/10.1016/j.neuroimage.2023.120259) paper.
![image](https://github.com/simnibs/simnibs/assets/29161453/edeea1d5-1a04-4c05-a256-eb7d3ec19c84)
How do I obtain the info.fif and trans.fif file?
For the info.fif, should I just make an mne Info object with my desired eeg cap?
For trans.fif, is this something I can get from a substep of simNIBS or should I just make it using [mne.gui.coregistration()](https://mne.tools/stable/generated/mne.gui.coregistration.html#mne.gui.coregistration)?

Would appreciate any help with both the current issue, and my potential steps list, or general feedback! Thanks!

**Comment by jduemose** [comment]
Hi, yes your approach sounds fine!

`info.fif`: Yes, you can save the info part of your data object separately or just pass the data itself (e.g., raw, epochs, etc.). Only the info part is read. As you correctly state, it should contain information about the electrodes.

`trans.fif`: You cannot generate this with simnibs, no, but you can definitely use MNE! I have not used the GUI myself, though, but it seems to be the tool you want. They also have [this](https://mne.tools/stable/auto_tutorials/forward/25_automated_coreg.html) tutorial. Seems what you want is `coreg.trans`.

trans is needed because MNE stores EEG positions in "head coordinates" whereas we want them in the space of the MR image (what I believe they call scanner RAS).

With regards to step 5 I am not sure I follow. Do you want to transform the leadfield to a standard space like fsaverage or you just want to sum leadfield vectors that belong to the same anatomical area?

Jesper

**Comment by anujanegi** [comment]
Thanks Jesper!
For `trans.fif`, to use MNE (`coreg.trans`), I would have to create a head model using Freesurfer first.
This defeats the purpose of using CHARM, with the usecase of leadfield generation.

Regarding step 5: yes, the prior. I want to sum leadfield vectors that belong to the same anatomical area according to a parcellation.

**Reply by jduemose** [reply]
I see, yeah the integration with freesurfer is pretty pervasive. However, in order to estimate the transformation, it just does (`estimate_head_mri_t`):
```
 from .channels.montage import make_dig_montage, compute_native_head_t

 subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
 lpa, nasion, rpa = get_mni_fiducials(subject, subjects_dir)
 montage = make_dig_montage(
     lpa=lpa["r"], nasion=nasion["r"], rpa=rpa["r"], coord_frame="mri"
 )
 return invert_transform(compute_native_head_t(montage))
```
So you need to get the position of the fiducials in MRI space (perhaps you have those already). You can manually identify or even use the warped (from MNI) fiducials in the `eeg_positions` subfolder of your charm run (fiducials.csv). (This is similar to what they do here by calling `get_mni_fiducials`.) These are in MRI space (in MNE, coord frame "mri" actually refers to freesurfer's mri space but this is a detail and we just pretend this is the coord frame that our MRI is in). Use these to create the DigMontage with keys as above and call `invert_transform(compute_native_head_t(montage))`.

Hopefully this will work and give you a head to mri transform. (the eeg positions in the data object should already have been converted head coord frame - automatically I guess).

Let me know if you have any problems.
